starfish/image/_segmentation/watershed.py

```diff
@@ -12,15 +12,37 @@
 from starfish.imagestack.imagestack import ImageStack
 from starfish.munge import relabel
 from starfish.stats import label_to_regions
-from starfish.types import Indices
+from starfish.types import Indices, Number
 from ._base import SegmentationAlgorithmBase
 
 
 class Watershed(SegmentationAlgorithmBase):
-    """
-    Implements watershed segmentation.  TODO: (dganguli) FILL IN DETAILS HERE PLS.
-    """
-    def __init__(self, dapi_threshold, input_threshold, min_distance, **kwargs) -> None:
+
+    def __init__(
+        self,
+        dapi_threshold: Number,
+        input_threshold: Number,
+        min_distance: int,
+        **kwargs
+    ) -> None:
+        """Implements watershed segmentation of cells seeded from a DAPI image the the point clouds
+
+        Watershed segmentation proceeds by constructing basins that extend from the DAPI image
+        and the point cloud constructed from a maximum projection across rounds and channels. It
+        constructs a mask that extends from the basins to prevent non-cellular regions from being
+        included.
+
```

Comment:
add: this simply wraps skimage. blah blah blah dot watershed and link to that functions documentation as well?